airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -69,7 +73,14 @@ def add_legacy_format(schema: dict) -> dict:
         config must be adjusted to support the generic mapping object. Once configs no longer adhere to the old
         format we can remove this change.
         """
+        legacy_format_options = {
```

Comment:
Originally the `{"type": "object"}` to represent the legacy type worked when the restriction was `anyOf`. But now that we need to match exactly one with `oneOf`, we need to be more granular with this check. The legacy s3 format is always indicated with the `filetype`. We have existing tests scenarios in both formats to verify this works